packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -34,6 +34,8 @@ abstract class RunCommandBase extends FlutterCommand {
               'forwards the host port to a device port.');
     argParser.addOption('route',
         help: 'Which route to load when running the app.');
+    argParser.addOption('target-platform',
```

Comment:
Maybe we should explain in help what happens when target platform is android-arm64 and the device is an ios or an 32 bit android device?

Reply:
Added some help text